# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/StoragePrincipalHelper.h
# Commit: 1a38b75a7dbb
# Full Hash: 1a38b75a7dbb86fa14d2763e7a983dfd0bde5d50
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2021-01-22 10:17:55
# Regressor Bug: 1686828
# File Overlap Count: 1
# Description:
#   Bug 1686828 - Use https OriginAttributes to fetch HTTPS RR r=necko-reviewers,timhuang,dragana
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D102234
# ==============================================================================

diff -r e83b6fb52b62 -r 1a38b75a7dbb toolkit/components/antitracking/StoragePrincipalHelper.h
--- a/toolkit/components/antitracking/StoragePrincipalHelper.h	Thu Jan 21 22:08:27 2021 +0000
+++ b/toolkit/components/antitracking/StoragePrincipalHelper.h	Thu Jan 21 22:08:26 2021 +0000
@@ -275,6 +275,14 @@
   // For HSTS we want to force 'HTTP' in the partition key.
   static bool GetOriginAttributesForHSTS(nsIChannel* aChannel,
                                          OriginAttributes& aAttributes);
+
+  // Like the function above, this function forces `HTTPS` in the partition key.
+  // The OA created by this function is mainly used in DNS cache. The spec
+  // specifies that the presence of HTTPS RR for an origin also indicates that
+  // all HTTP resources are available over HTTPS, so we use this function to
+  // ensure that all HTTPS RRs in DNS cache are accessed by HTTPS requests only.
+  static bool GetOriginAttributesForHTTPSRR(nsIChannel* aChannel,
+                                            OriginAttributes& aAttributes);
 };
 
 }  // namespace mozilla
